.github/workflows/additional-ci-image-checks.yml

```
@@ -93,7 +96,7 @@ jobs:
       packages: write
     secrets: inherit
     with:
-      # Runs on Public runners
+      runs-on-as-string: ${{ inputs.public-runs-on-as-string }}
```

```suggestion
      public-runs-on-as-string: ${{ inputs.public-runs-on-as-string }}
```
Just wondering if this would make it more consistent(seems this is the only place without the public/default prefix) and explicit.

---

yeah. It was another (a bit) of inconsistency. I just thought that in case of those two workflows (pushing cache and building images) we always know which runner we want to use, and I wanted to pass the right one). Those workflows could be executed with `public` or `self-hosted` depending on the platform (AMD/ ARM). I just wanted to avoid pushing both as parameters and making the choice in the final workflow, but I think yeah, this is the right approach - so I pushed a change with this ternary down to the "leaf" workflows - as this will make run-as parameters even more explicit and consistent.

```
      ${{ inputs.platform == 'linux/amd' && fromJSON(inputs.runs-on-as-json-public) ||
      fromJSON(inputs.runs-on-as-json-self-hosted) }}
```

---

Should be much more consistent now @pankajkoti @hussein-awala

---

Actually - it seems that it has to be this way:

```
    runs-on: >
      ${{ fromJSON(inputs.platform == 'linux/amd' && inputs.runs-on-as-json-public ||
            inputs.runs-on-as-json-self-hosted) }}
```

---

🤷  with GitHub Actions ternary expressions

---

Still have not figured out if it works or not and how :)